## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 434552`

Logs: https://github.com/qbisi/nixpkgs-review-gha/actions/runs/17108490625


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>elmerfem</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>elmerfem</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>elmerfem</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>elmerfem</summary>
<pre>


CMake Error in ElmerGUI/Application/CMakeLists.txt:
  Found relative path while evaluating include directories of "ElmerGUI":

    "QWT_INCLUDE_DIR-NOTFOUND"



-- Generating done (7.0s)
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_OpenGL_GL_PREFERENCE
    CMAKE_POLICY_DEFAULT_CMP0025


CMake Generate step failed.  Build files cannot be regenerated correctly.</pre>
</details>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>elmerfem</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>elmerfem</summary>
<pre>


CMake Error in ElmerGUI/Application/CMakeLists.txt:
  Found relative path while evaluating include directories of "ElmerGUI":

    "QWT_INCLUDE_DIR-NOTFOUND"



-- Generating done (1.1s)
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_OpenGL_GL_PREFERENCE
    CMAKE_POLICY_DEFAULT_CMP0025


CMake Generate step failed.  Build files cannot be regenerated correctly.</pre>
</details>
</details>

--- a/pkgs/by-name/el/elmerfem/package.nix
+++ b/pkgs/by-name/el/elmerfem/package.nix
@@ -11,9 +11,9 @@
   libGL,
   libGLU,
   opencascade-occt,
-  libsForQt5,
+  qt6Packages,
   tbb,
-  vtkWithQt5,
+  vtkWithQt6,
   llvmPackages,
 }:
 stdenv.mkDerivation rec {
@@ -33,21 +33,20 @@ stdenv.mkDerivation rec {
     cmake
     gfortran
     pkg-config
-    libsForQt5.wrapQtAppsHook
+    qt6Packages.wrapQtAppsHook
   ];
 
   buildInputs = [
     mpi
     blas
     liblapack
-    libsForQt5.qtbase
-    libsForQt5.qtscript
-    libsForQt5.qwt
+    qt6Packages.qtbase
+    qt6Packages.qwt
     libGL
     libGLU
     opencascade-occt
     tbb
-    vtkWithQt5
+    vtkWithQt6
   ]
   ++ lib.optional stdenv.cc.isClang llvmPackages.openmp;
 
@@ -68,7 +67,10 @@ stdenv.mkDerivation rec {
     (lib.cmakeFeature "CMAKE_INSTALL_LIBDIR" "lib")
     (lib.cmakeFeature "CMAKE_INSTALL_INCLUDEDIR" "include")
     (lib.cmakeFeature "CMAKE_OpenGL_GL_PREFERENCE" "GLVND")
+  ]
+  ++ lib.optionals stdenv.hostPlatform.isDarwin [
     (lib.cmakeBool "USE_MACOS_PACKAGE_MANAGER" false)
+    (lib.cmakeFeature "QWT_INCLUDE_DIR" "${qt6Packages.qwt}/lib/qwt.framework/Headers")
   ];
 
   meta = with lib; {

--- a/pkgs/by-name/el/elmerfem/package.nix
+++ b/pkgs/by-name/el/elmerfem/package.nix
@@ -55,22 +55,20 @@ stdenv.mkDerivation rec {
     patchShebangs ./
   '';
 
-  storepath = placeholder "out";
-
   NIX_CFLAGS_COMPILE = "-Wno-error=implicit-function-declaration";
 
   cmakeFlags = [
-    "-DELMER_INSTALL_LIB_DIR=${storepath}/lib"
-    "-DWITH_OpenMP:BOOLEAN=TRUE"
-    "-DWITH_MPI:BOOLEAN=TRUE"
-    "-DWITH_QT5:BOOLEAN=TRUE"
-    "-DWITH_OCC:BOOLEAN=TRUE"
-    "-DWITH_VTK:BOOLEAN=TRUE"
-    "-DWITH_ELMERGUI:BOOLEAN=TRUE"
-    "-DCMAKE_INSTALL_LIBDIR=lib"
-    "-DCMAKE_INSTALL_INCLUDEDIR=include"
-    "-DCMAKE_OpenGL_GL_PREFERENCE=GLVND"
-    "-DUSE_MACOS_PACKAGE_MANAGER=False"
+    (lib.cmakeFeature "ELMER_INSTALL_LIB_DIR" "${placeholder "out"}/lib")
+    (lib.cmakeBool "WITH_OpenMP" true)
+    (lib.cmakeBool "WITH_MPI" true)
+    (lib.cmakeBool "WITH_QT6" true)
+    (lib.cmakeBool "WITH_OCC" true)
+    (lib.cmakeBool "WITH_VTK" true)
+    (lib.cmakeBool "WITH_ELMERGUI" true)
+    (lib.cmakeFeature "CMAKE_INSTALL_LIBDIR" "lib")
+    (lib.cmakeFeature "CMAKE_INSTALL_INCLUDEDIR" "include")
+    (lib.cmakeFeature "CMAKE_OpenGL_GL_PREFERENCE" "GLVND")
+    (lib.cmakeBool "USE_MACOS_PACKAGE_MANAGER" false)
   ];
 
   meta = with lib; {

